src/store/modules/ProfilesModule.ts

```diff
@@ -0,0 +1,51 @@
+import FileUtils from "../../utils/FileUtils";
+import R2Error from "../../model/errors/R2Error";
+import { ActionTree, Store } from "vuex";
+import { State as RootState } from "../../store";
+import Profile from "../../model/Profile";
+import FsProvider from "../../providers/generic/file/FsProvider";
+
+interface State {
+    profileList: string[];
+}
+
+/**
+ * State for Profiles, i.e. list for profiles in a single game/community.
+ */
+export const ProfilesModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        profileList: ['Default'],
+    }),
+    mutations: {
+        setProfileList(state: State, profileList: string[]) {
+            state.profileList = profileList;
+        },
+    },
+    actions: <ActionTree<State, RootState>>{
+        async removeSelectedProfile({commit, rootGetters, state, dispatch}, params: {fs: FsProvider, store: Store<any>}) {
+            const activeProfile: Profile = rootGetters['profile/activeProfile'];
+            const path = activeProfile.getPathOfProfile();
+            const profileName = activeProfile.getProfileName();
+
+            try {
```

Comment:
So I probably didn't mention this earlier, but I've been using an approach where the Vuex-methods just throw R2Errors (or just plain JS errors if we don't know what errors to expect and don't know what to use as name/suggestion parameters. The UI components then try-catch the operations they call and dispatch the `error/handleError`.

The reason is more philosophical than technical. I see showing the error modal as an UI concern, so it should be handled by the UI component. This (and other) Vuex actions are more business logic, and don't care that much about how the stuff is shown on UI.

So for consistency, use that approach here and in the actions you create in the future.